code/game/machinery/constructable_frame.dm

```diff
@@ -1260,3 +1260,10 @@ obj/item/weapon/circuitboard/rdserver
 	name = "Circuit Board (Isotope Ratio Spectrometer)"
 	desc = "A circuit board used to run a machine used in xenoarcheology."
 	build_path = "/obj/machinery/anomaly/isotope_ratio"
+
+/obj/item/weapon/circuitboard/confectionator
+
+	name = "circuit board (confectionator)"
+	desc = "A circuit board used to run a kitchen appliance."
+	board_type = MACHINE
+	build_path = "/obj/machinery/cooking/deepfryer/confectionator"
```

Comment:
You need to set the `req_components` in this type so it knows what part the construction requires.